curriculum/challenges/english/07-scientific-computing-with-python/learn-classes-and-objects-by-building-a-sudoku-solver/660a937220bf966fd844f1ee.md

```diff
@@ -0,0 +1,88 @@
+---
+id: 660a937220bf966fd844f1ee
+title: Step 64
+challengeType: 20
+dashedName: step-64
+---
+
+# --description--
+
+Now, add a `print()` call to print the following f-string: `f'Puzzle to solve:\n{gameboard}'`.
+
+# --hints--
+
+You should print the provided string inside the `solve_sudoku` function.
+
+```js
+({ test: () => assert(runPython(`_Node(_code).find_function("solve_sudoku").has_call("print(f'Puzzle to solve:\\n{gameboard}')")`)) })
```

Comment:
```suggestion
({ test: () => assert(runPython(`_Node(_code).find_function("solve_sudoku").has_call("print(f'Puzzle to solve:\\\\n{gameboard}')")`)) })
```
It's awkward, but remember that the python string has to be `"print(f'Puzzle to solve:\\n{gameboard}')" so that `\n` isn't interpreted as a newline. Then we need to put this inside a JS string, so each slash must be escaped again.